src/thor/costmatrix.cc

```diff
@@ -357,6 +357,9 @@ void CostMatrix::ForwardSearch(const uint32_t index, const uint32_t n, GraphRead
 
     // Handle transitions - expand from the end node of the transition
     if (!from_transition && nodeinfo->transition_count() > 0) {
+      if (tile == nullptr) {
```

Comment:
this is not needed. you'll notice that before each call to expand, there is an if that protects from calling it with a null tile